### PR TITLE
V3—2D_dot_chart

### PR DESCRIPTION
**Pull request by bfinzer**
Given a dot chart, dropping a categorical attribute on the empty axis splits the dot chart by the new categories.
* Note that the stack of dots grows perpendicular to the _first_ attribute and that therefore the stacks can be made to grow either vertically _or_ horizontally depending on which attribute was first. There are some subtleties brought about by _replacing_ the first attribute with a different categorical which currently forces the growth direction to switch. We may want to change this behavior.
* As with split dot plots the grid lines remain to be implemented

### CODE REVIEW

File: v3/src/components/graph/components/chartdots.tsx
Comment (kswenson) [inline v3/src/components/graph/components/chartdots.tsx:56]: Use the `const { cases } = dataConfig` trick above so that this just depends on `dataConfig`.

File: v3/src/components/graph/components/chartdots.tsx
Comment (kswenson) [inline v3/src/components/graph/components/chartdots.tsx:77]: We haven't talked about this before, and it need not hold up this PR, but it's worth considering whether a JS `Map` would make sense in some of these places where we're using a simple JS object as a map.

File: v3/src/components/graph/models/data-configuration-model.ts
Comment (kswenson) [inline v3/src/components/graph/models/data-configuration-model.ts:109]: There should be some tests of these methods as well, but I leave it to your discretion whether that's a task for this PR or not.

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Generally looks good -- just a few suggestions. Also, there's a lint warning for an undeclared `useEffect` dependency in `DotPlotDots` that's left over from the previous PR that would be good to fix.

File: v3/src/components/graph/components/chartdots.tsx
Comment (bfinzer) [reply v3/src/components/graph/components/chartdots.tsx:77]: Interesting. There certainly have been times in V2 when I had to work around the inability to guarantee stable insertion order.

File: v3/src/components/graph/models/data-configuration-model.ts
Comment (bfinzer) [reply v3/src/components/graph/models/data-configuration-model.ts:109]: I'll include tests in this PR. Also, I think the detection/elimination of empty values should occur in attributeValuesForPlace, obviating the need to delete them from the Set.

Comment (bfinzer) [reply v3/src/components/graph/models/data-configuration-model.ts:109]: I've added them now.

**Review by kswenson — APPROVED** [review]
👍 Changes look good

**Pull request merged**
